include/my_global.h

```diff
@@ -530,7 +530,7 @@ typedef double	pfloat;		/* Mixed prototypes can't take float */
 #endif
 C_MODE_START
 typedef int	(*qsort_cmp)(const void *,const void *);
-typedef int	(*qsort_cmp2)(void*, const void *,const void *);
+typedef int	(*qsort_cmp2)(const void *param, const void *a, const void *b);
```

Comment:
What about for uses of the `const`-defined `qsort2_cmp` which pass in variables that are originally defined as `const` from the start? For example, comparisons using `CHARSET_INFO *` (derives to `const charset_info_st *`), as done in `sql/item_cmpfunc.h`'s `in_vector::compare_elems`? It doesn't seem good to cast away the `const` qualifier, and keeping `const` seems more appropriate for a comparison.

Reply:
I think that casting away `const` is only acceptable when the `const` will not be violated. Currently, there are some places where we are actually modifying the non-`const` pointer target, such as `rowid_cmp_reverse()`. That is exactly like the first ‘bad’ example [in the C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es50-dont-cast-away-const).

It would feel more ‘honest’ to not declare the context pointer without `const`.